plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -388,7 +393,18 @@ def sweep(self, arguments: List[str]) -> None:
                 except Exception as e:
                     state = optuna.trial.TrialState.FAIL
                     study.tell(trial=trial, state=state, values=values)
-                    raise e
+                    log.warning(f"Failed experiment: {e}")
+                    failures.append(e)
+
+            # raise if too many failures
+            if len(failures) / len(returns) > self.max_failure_rate:
+                log.error(
+                    f"Failed {failures} times out of {len(returns)} "
+                    f"with max_failure_rate={self.max_failure_rate}."
+                )
+                assert len(failures) > 0
+                # Raise the first failure
+                raise failures[0]
```

Comment:
overall looks great! Thanks so much for working on this!

could you follow the [nevergrad example here](https://github.com/facebookresearch/hydra/pull/1545/files#diff-5acb34ece4bf6c5b46d41fcdfe8da4abff87dc867221145df01462ff75da15e7R188)? 

This will let Hydra handle the exception, the difference running the example app is the following
## current version
```
[2022-05-26 14:20:26,340][HYDRA] Failed [RuntimeError('cfg.error is True')] times out of 1 with max_failure_rate=0.0.
Traceback (most recent call last):
  File "/Users/jieru/workspace/hydra/plugins/hydra_optuna_sweeper/example/sphere.py", line 12, in sphere
    raise RuntimeError("cfg.error is True")
RuntimeError: cfg.error is True

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.

```

### nevergrad version
```
[2022-05-26 14:26:02,117][HYDRA] Failed [RuntimeError('cfg.error is True')] times out of 1 with max_failure_rate=0.0.
Error executing job with overrides: ['x=2.5', 'y=5', 'error=True']
Traceback (most recent call last):
  File "/Users/jieru/workspace/hydra/plugins/hydra_optuna_sweeper/example/sphere.py", line 12, in sphere
    raise RuntimeError("cfg.error is True")
RuntimeError: cfg.error is True

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.

```

Hydra will provide a bit extra detail, this will also ensure if in future we improve Hydra's exception handing, optuna will benefit as well.

Reply:
Should be fixed now :)